crates/ruff_linter/src/rules/ruff/rules/if_key_in_dict_del.rs

```diff
@@ -0,0 +1,182 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Applicability, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::comparable::ComparableExpr;
+use ruff_python_ast::{
+    CmpOp, Expr, ExprCompare, ExprName, ExprSubscript, Stmt, StmtDelete, StmtIf,
+};
+use ruff_python_semantic::analyze::typing;
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::checkers::ast::Checker;
+
+// Real type: Expr::Name;
+type Key = Expr;
```

Comment:
I'd suggest to create an enum for the `Key`. It makes it easier to reason about the rule and gives you a good place to implement some operations (e.g. implement `Eq` manually)

```
enum Key<'ast> {
	Name(&'ast ExprName),
	None,
	Ellipsis, 
	StringLiteral(&'ast StringLiteral),
	BytesLiteral(&'ast BytesLiteral)
}
```

Reply:
You can also store all nodes if you need the ranges